Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java

```diff
@@ -34,92 +54,162 @@
-                String jsonData = jArray.toString(); 
-                model.addAttribute("jsonstring", jsonData);
-                model.addAttribute("networkid", "\"\"");
-
-            } catch (JSONException e) {
-
-                StringBuffer errorMsg = new StringBuffer();
-                model.addAttribute("show_error_alert", true);
-                errorMsg.append(e.getMessage());
-                errorMsg.append("\n");
-                model.addAttribute("error_alert_msg", errorMsg.toString());
-            }
-        }
-        model.addAttribute("project", project);
-        return "sites/project/statistics";
-    }
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	@Autowired
+	private INetworkManager networkmanager;
+
+	@Autowired
+	private ProjectStats projectStats;
+
+	@Autowired
+	private Environment env;
+
+	@Autowired
+	private IEditorManager editorManager;
+
+	@Autowired
+	private IUserManager userManager;
+
+	@Autowired
+	private IProject iprojectManager;
+
+	@Autowired
+	private IProjectWorkspace workspace; 
+
+	private static final Logger logger = LoggerFactory.getLogger(ProjectStatsController.class);
+	private static final String SUBMITTED = "SUBMITTED";
+	private static final String APPROVED = "APPROVED";
+	private static final String REJECTED = "REJECTED";
+
+
+
+	private int getCount(List<IConceptStats> conceptList) {
+		int cnt = Integer.parseInt(env.getProperty("project.stats.topcount"));
+		int len = conceptList.size() > cnt ? cnt : conceptList.size();
+		return len;
+	}
+
+	private JSONArray getTopConceptsJson(List<IConceptStats> conceptsList,
+			int length) throws JSONException {
+		JSONArray jsonArray = new JSONArray();
+
+		for (int i = 0; i < length; i++) {
+			JSONObject jsonObject = new JSONObject();
+			IConceptStats conceptStats = conceptsList.get(i);
+			jsonObject.put("conceptId", conceptStats.getConceptId().replace("\"", ""));
+			jsonObject.put("description", conceptStats.getDescription().replace("\"", ""));
+			jsonObject.put("label", conceptStats.getLemma().replace("\"", ""));
+			jsonObject.put("count", conceptStats.getCount());
+			jsonArray.put(jsonObject);
+		}
+		return jsonArray;
+	}
+
+	/**
+	 * This method returns Json Array containing date and corresponding count of number 
+	 * of networks that were Approved/Rejected/Submitted depending on 'status' 
+	 * @author Bharath Srikantan
+	 * @throws JSONException 
+	 *
+	 */
+	private JSONArray getContributionCountByStatus (List<INetwork> networks, String status) throws JSONException
+	{
+		String DATE_FORMAT = "dd-MMM-yy";
+		SimpleDateFormat sdf = new SimpleDateFormat(DATE_FORMAT);
+		JSONArray contributionsJson = new JSONArray();
+		HashMap<String, Integer> contributionCount = new HashMap<String, Integer>();
+
+		for(INetwork network : networks)
+		{
+			if(network.getStatus().equals(status) || status.equals("SUBMITTED"))
+			{
+				if(contributionCount.containsKey(sdf.format(network.getCreatedDate())))

```

Comment:
You should assign this ```sdf.format(network.getCreatedDate())``` once into a variable, and then use the variable instead of repeating it over and over again.